datasette/static/cm-editor.js

```diff
@@ -0,0 +1,46 @@
+import { EditorView, basicSetup } from "codemirror";
+import { keymap } from "@codemirror/view";
+import { sql, SQLite } from "@codemirror/lang-sql";
+
+// Utility function from https://codemirror.net/docs/migration/
+export function editorFromTextArea(textarea) {
+  // This could also be configured with a set of tables and columns for better autocomplete:
+  // https://github.com/codemirror/lang-sql#user-content-sqlconfig.tables
+  let view = new EditorView({
+    doc: textarea.value,
+    extensions: [
+      basicSetup,
+      EditorView.lineWrapping,
+      keymap.of([
+        {
+          key: "Shift-Enter",
+          run: function () {
+            textarea.value = view.state.doc.toString();
+            textarea.form.submit();
+          },
+          preventDefault: true,
+        },
+      ]),
+      sql({
+        dialect: SQLite,
+      }),
+    ],
+  });
+
+  // Idea taken from https://discuss.codemirror.net/t/resizing-codemirror-6/3265.
```

Comment:
It's unclear to me if this ResizeObserver and requestMeasure is needed. It's suggested in the thread but from some light manual testing the behavior seems the same with or without

Reply:
For manual testing, any changes to this file require the build as outlined in the docs:

```
node_modules/.bin/rollup datasette/static/cm-editor.js -f iife -n cm -o datasette/static/cm-editor.bundle.js \
       -p @rollup/plugin-node-resolve -p @rollup/plugin-terser
```